sdk/azcore/log.go

```diff
@@ -31,86 +30,34 @@ const (
 )
 
-// Listener is the function signature invoked when writing log entries.
-// A Listener is required to perform its own synchronization if it's
-// expected to be called from multiple Go routines.
-type Listener func(LogClassification, string)
-
-// Logger controls which classifications to log and writing to the underlying log.
-type Logger struct {
-	cls []LogClassification
-	lst Listener
-}
-
-// SetClassifications is used to control which classifications are written to
-// the log.  By default all log classifications are written.
-func (l *Logger) SetClassifications(cls ...LogClassification) {
-	l.cls = cls
-}
-
-// SetListener will set the Logger to write to the specified Listener.
-func (l *Logger) SetListener(lst Listener) {
-	l.lst = lst
-}
-
-// Should returns true if the specified log classification should be written to the log.
-// By default all log classifications will be logged.  Call SetClassification() to limit
-// the log classifications for logging.
-// If no listener has been set this will return false.
-// Calling this method is useful when the message to log is computationally expensive
-// and you want to avoid the overhead if its log classification is not enabled.
-func (l *Logger) Should(cls LogClassification) bool {
-	if l.lst == nil {
-		return false
-	}
-	if l.cls == nil || len(l.cls) == 0 {
-		return true
+func SetClassifications(cls ...LogClassification) {
+	input := make([]logger.LogClassification, 0)
+	for _, l := range cls {
+		input = append(input, logger.LogClassification(l))
 	}
-	for _, c := range l.cls {
-		if c == cls {
-			return true
-		}
-	}
-	return false
+	logger.Log().SetClassifications(input...)
 }
 
-// Write invokes the underlying Listener with the specified classification and message.
-// If the classification shouldn't be logged or there is no listener then Write does nothing.
-func (l *Logger) Write(cls LogClassification, message string) {
-	if !l.Should(cls) {
-		return
-	}
-	l.lst(cls, message)
-}
+type Listener func(LogClassification, string)
 
-// Writef invokes the underlying Listener with the specified classification and formatted message.
-// If the classification shouldn't be logged or there is no listener then Writef does nothing.
-func (l *Logger) Writef(cls LogClassification, format string, a ...interface{}) {
-	if !l.Should(cls) {
-		return
+func transform(lst Listener) logger.Listener {
+	return func(l logger.LogClassification, msg string) {
+		azcoreL := LogClassification(l)
+		lst(azcoreL, msg)
 	}
-	l.lst(cls, fmt.Sprintf(format, a...))
 }
 
-// for testing purposes, nolint is a false positive
-func (l *Logger) resetClassifications() { //nolint:unused
-	l.cls = nil
-}
-
-var logger Logger
-
-// Log returns the process-wide logger.
-func Log() *Logger {
-	return &logger
+func SetListener(lst Listener) {
+	if lst == nil {
+		fmt.Println("nil listener")
+		logger.Log().SetListener(nil)
+	} else {
+		fmt.Println("Not nil listener")
+		logger.Log().SetListener(transform(lst))
```

Comment:
You can just put the code that's in transform() directly here in `SetListener()`

Reply:
I added a doc comment for future Sean to understand what the transform does and why it is there, I'd like to keep it as a separate function for readability.